As each English Premier League season finishes, it seems as though the top teams spend more and more money to get the best players the world has to offer. Players like Harry Kane, Kevin De Bruyne, and Mohamed Salah are all worth over 100 million dollars, and some teams are willingto spend even more for young talent. This begs the question, does spending more money on players lead to better results?

The main goal of this project is to see if there is a correlation between how much a soccer team in the English Premier League (EPL) spends on players and how well they perform in the league. The data used in this project starts in the 2017 season and goes to current. player acquisitions are a huge expense for a soccer club, so if a team could cut these costs and still perform well, it would be a huge advantage.

We will measure this first by finding how much money each club has spent on transfers in the last five years. Next is to give each team a score based on how well they have performed in the last five years. If a team won the league, they would receive 5 points. Qualifying for the champions league (finishing in the top 4) would be 3 points and finally qualifying for the europa league (positions 5-7) will receive 1 point. It is important to note that we are only measuring success based on performance in the EPL; domestic tournaments, champions league results and eurpa league results do not count towards a teams score.

 We can then compare how each team did over the 5 year span to see if the the clubs that spent more money performed better. This will give us a better idea of how much money a team should spend on players to perform well in the EPL. 
 
 We found a useful .csv file in Github that had the league table of the EPL from 2000-2022. To get information on expenditures, we scracped data from transfermarkt.us.

In [108]:

import pandas as pd
import numpy as np
import requests
from bs4 import BeautifulSoup

In [109]:
# Read in the data for standings from 2000-2021

df_2 = pd.read_csv("EPL Standings 2000-2022.csv")

In [110]:
#Scraping data from transfermarkt.us and creating a dataframe with the expenditures and income for each team

headers = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36'}


expenditures = "https://www.transfermarkt.us/premier-league/einnahmenausgaben/wettbewerb/GB1/plus/0?ids=a&sa=&saison_id=2017&saison_id_bis=2022&nat=&pos=&altersklasse=&w_s=&leihe=&intern=0"
expenditures = requests.get(expenditures, headers=headers)
soup = BeautifulSoup(expenditures.content, 'html.parser')

# Get the table
table = soup.find('table', {'class': 'items'})

# Get the table headers and remove all except the team names and the expenditures
table_headers = table.find_all('th')
table_headers = [th.text for th in table_headers]
table_headers = table_headers[1:21]
table_headers = table_headers[::2]
table_headers

# Get the table rows
table_rows = table.find_all('tr')
table_rows = [tr.text for tr in table_rows]
table_rows = table_rows[1:]
table_rows

# Clean the data
table_rows = [row.replace('\n', '') for row in table_rows]
table_rows = [row.replace('\t', '') for row in table_rows]
table_rows = [row.replace(' ', '') for row in table_rows]
table_rows

# Split the rows up by table headers
table_rows = [row.split('€') for row in table_rows]
table_rows

# Create a dataframe
df = pd.DataFrame(table_rows, columns=table_headers)

# remove the numbers from the team names
df['Club'] = df['Club'].str.replace('\d+', '')

# remove the last three characters from the expenditure and income columns
df['Expenditure'] = df['Expenditure'].str[:-3]
df['Income'] = df['Income'].str[:-3]
df = df.replace('', np.nan)

# remove the remaining letters from the expenditure and income and balance columns
df['Expenditure'] = df['Expenditure'].str.replace('[a-zA-Z]', '')
df['Income'] = df['Income'].str.replace('[a-zA-Z]', '')
df['Balance'] = df['Balance'].str.replace('[a-zA-Z]', '')

# convert the expenditure, income, and balance columns to a float
df['Expenditure'] = df['Expenditure'].astype(float)
df['Income'] = df['Income'].astype(float)
df['Balance'] = df['Balance'].astype(float)
df

# chage the first two value in the expenditure column to thousands
df['Expenditure'][0] = df['Expenditure'][0] * 1000
df['Expenditure'][1] = df['Expenditure'][1] * 1000
df

# add spaces to the team names
df['Club'] = df['Club'].str.replace('ManchesterCity', 'Manchester City')
df['Club'] = df['Club'].str.replace('ManchesterUnited', 'Manchester United')
df['Club'] = df['Club'].str.replace('TottenhamHotspur', 'Tottenham Hotspur')
df['Club'] = df['Club'].str.replace('WestHamUnited', 'West Ham United')
df['Club'] = df['Club'].str.replace('LeicesterCity', 'Leicester City')
df['Club'] = df['Club'].str.replace('NewcastleUnited', 'Newcastle United')
df['Club'] = df['Club'].str.replace('WolverhamptonWanderers', 'Wolverhampton Wanderers')
df['Club'] = df['Club'].str.replace('WestBromwichAlbion', 'West Bromwich Albion')
df['Club'] = df['Club'].str.replace('SheffieldUnited', 'Sheffield United')
df['Club'] = df['Club'].str.replace('Brighton&HoveAlbion', 'Brighton & Hove Albion')
df['Club'] = df['Club'].str.replace('AstonVilla', 'Aston Villa')
df['Club'] = df['Club'].str.replace('CrystalPalace', 'Crystal Palace')
df['Club'] = df['Club'].str.replace('LeedsUnited', 'Leeds United')

# remove the letters FC from the team names
df['Club'] = df['Club'].str.replace('FC', '')

# remove the letter A at the beginning of ABournemouth
df['Club'] = df['Club'].str.replace('ABournemouth', 'Bournemouth')

df


/var/folders/k9/_qdn4hln3xs71fjm8zsm8g3c0000gn/T/ipykernel_654/765690108.py:40: FutureWarning: The default value of regex will change from True to False in a future version.
  df['Club'] = df['Club'].str.replace('\d+', '')
/var/folders/k9/_qdn4hln3xs71fjm8zsm8g3c0000gn/T/ipykernel_654/765690108.py:48: FutureWarning: The default value of regex will change from True to False in a future version.
  df['Expenditure'] = df['Expenditure'].str.replace('[a-zA-Z]', '')
/var/folders/k9/_qdn4hln3xs71fjm8zsm8g3c0000gn/T/ipykernel_654/765690108.py:49: FutureWarning: The default value of regex will change from True to False in a future version.
  df['Income'] = df['Income'].str.replace('[a-zA-Z]', '')
/var/folders/k9/_qdn4hln3xs71fjm8zsm8g3c0000gn/T/ipykernel_654/765690108.py:50: FutureWarning: The default value of regex will change from True to False in a future version.
  df['Balance'] = df['Balance'].str.replace('[a-zA-Z]', '')
/var/folders/k9/_qdn4hln3xs71fjm8zsm8g3c0000gn/T/ipykernel_654/765690

Club  Expenditure  Income  Balance
0                   Chelsea      1490.00  684.70  -806.29
1           Manchester City      1030.00  539.92  -485.79
2         Manchester United       984.98  220.68  -764.30
3                   Arsenal       839.56  299.10  -540.46
4                 Liverpool       674.60  397.77  -276.83
5         Tottenham Hotspur       655.80  260.32  -395.48
6                   Everton       615.07  396.86  -218.22
7           West Ham United       600.70  209.46  -391.24
8   Wolverhampton Wanderers       556.36  227.59  -328.77
9          Newcastle United       534.00  110.48  -423.52
10              Aston Villa       512.69  220.75  -291.94
11           Leicester City       477.75  373.57  -104.17
12              Southampton       409.05  243.90  -165.15
13   Brighton & Hove Albion       374.59  259.52  -115.07
14             Leeds United       372.81  169.70  -203.11
15                   Fulham       303.42   79.45  -223.97
16              Bournemouth       276.55  175.28  -101.28
17         NottinghamForest       249.07   65.98  -183.09
18           Crystal Palace       223.02   84.18  -138.84
19                  Watford       221.81  237.16    15.35
20                  Burnley       181.29  190.81     9.52
21         Sheffield United       150.34   40.75  -109.59
22                Brentford       148.02  150.98     2.96
23     West Bromwich Albion       142.31   95.10   -47.21
24                StokeCity       134.15  101.59   -32.56

In [134]:

df_2 = df_2.tail(100)

df_2

# Drop the second year in the season column
df_2['Season'] = df_2['Season'].str[:4]

df_2


Season  Pos               Team  Pld   W   D   L   GF  GA  GD  Pts  \
340   2017    1    Manchester City   38  32   4   2  106  27  79  100   
341   2017    2  Manchester United   38  25   6   7   68  28  40   81   
342   2017    3  Tottenham Hotspur   38  23   8   7   74  36  38   77   
343   2017    4          Liverpool   38  21  12   5   84  38  46   75   
344   2017    5            Chelsea   38  21   7  10   62  38  24   70   
..     ...  ...                ...  ...  ..  ..  ..  ...  ..  ..  ...   
435   2021   16            Everton   38  11   6  21   43  66 -23   39   
436   2021   17       Leeds United   38   9  11  18   42  79 -37   38   
437   2021   18            Burnley   38   7  14  17   34  53 -19   35   
438   2021   19            Watford   38   6   5  27   34  77 -43   23   
439   2021   20       Norwich City   38   5   7  26   23  84 -61   22   

                           Qualification or relegation  
340  Qualification for the Champions League group s...  
341  Qualification for the Champions League group s...  
342  Qualification for the Champions League group s...  
343  Qualification for the Champions League group s...  
344  Qualification for the Europa League group stag...  
..                                                 ...  
435                                     Not Applicable  
436                                     Not Applicable  
437                 Relegation to the EFL Championship  
438                 Relegation to the EFL Championship  
439                 Relegation to the EFL Championship  

[100 rows x 12 columns]

In [168]:
# Create a function that returns if the team was in the premier league in every year
def in_premier_league(df_2, team, year):
    if df_2.loc[(df_2['Team'] == team) & (df_2['Season'] == year), 'Team'].empty:
        return "No"
    else:
        return "Yes"
    
in_premier_league(df_2, 'Sheffield United', "2021")


'No'

In [179]:
# Use the in_premier_league function to create a function that returns if the team was never relegated
def never_relegated(df_2, team):
    for i in range(2017, 2022):
        if in_premier_league(df_2, team, str(i)) == "No":
            return "No"
    return "Yes"

never_relegated(df_2, 'Norwich City')

'No'

In [181]:
# Create a function that returns the total expenditure for a team
def total_expenditure(df, team_name):
    total = 0
    for i in range(len(df)):
        if df['Club'][i] == team_name:
            total += float(df['Expenditure'][i])
    return total

total_expenditure(df, 'Liverpool')

674.6

In [156]:
# Create a function that returns the league position for a team in a given year
def league_position(df_2, team, year):
    return df_2.loc[(df_2['Team'] == team) & (df_2['Season'] == year), 'Pos'].values[0]

league_position(df_2, 'Sheffield United', "2020")


    

20

In [149]:
# Create a function that returns if the team won the league in a given year
def won_league(df_2, team, year):
    if league_position(df_2, team, year) == 1:
        return "Yes"
    else:
        return "No"

won_league(df_2, 'Sheffield United', "2020")  
    

'No'

In [159]:
# Create a function that returns if the team qualified for the champions league in a given year
def qualified_for_champions_league(df_2, team, year):
    if league_position(df_2, team, year) <= 4:
        return "Yes"
    else:
        return "No"
    
qualified_for_champions_league(df_2, 'Manchester City', "2021")

'Yes'

In [196]:
# Create a function that returns if the team qualified for the europa league in a given year
def qualified_for_europa_league(df_2, team, year):
    if league_position(df_2, team, year) <= 7:
        if qualified_for_champions_league(df_2, team, year) == "No":
            return "Yes"
        else:
            return "No"
    else:
        return "No"
    
qualified_for_europa_league(df_2, 'Manchester City', "2021")

'No'

In [205]:
# Create a function that returns the performance of a team over the past five years
def performance_in_five_years(df, club):
    performance = 0
    if never_relegated(df_2, club) == "Yes":
        performance += 1 
        for year in range(2017, 2022):
            if won_league(df_2, club, str(year)) == "Yes":
                performance += 5
            elif qualified_for_champions_league(df_2, club, str(year)) == "Yes":
                performance += 3
            elif qualified_for_europa_league(df_2, club, str(year)) == "Yes":
                performance += 1
    else:
        return "N/A"
    return performance

performance_in_five_years(df, 'West Ham United')


        
               
    
    

3

In [206]:
#make a dictionary of the teams and their performance
teams = df['Club'].unique()
performance = {}
for team in teams:
    performance[team] = performance_in_five_years(df, team)

performance

# make a dataframe of the teams and their performance
df_3 = pd.DataFrame(list(performance.items()), columns=['Team', 'Performance'])

df_3








Team Performance
0                   Chelsea          14
1           Manchester City          24
2         Manchester United          12
3                   Arsenal           4
4                 Liverpool          18
5         Tottenham Hotspur          12
6                   Everton           1
7           West Ham United           3
8   Wolverhampton Wanderers         N/A
9          Newcastle United           1
10              Aston Villa         N/A
11           Leicester City           3
12              Southampton           1
13   Brighton & Hove Albion           1
14             Leeds United         N/A
15                   Fulham         N/A
16              Bournemouth         N/A
17         NottinghamForest         N/A
18           Crystal Palace           1
19                  Watford         N/A
20                  Burnley           2
21         Sheffield United         N/A
22                Brentford         N/A
23     West Bromwich Albion         N/A
24                StokeCity         N/A

In [207]:
# Create a new dataframe with the team names, expenditures, and performance in the last five years only 

df_4 = pd.DataFrame(columns=['Team', 'Expenditure', 'Performance'], index=range(len(df)))

for i in range(len(df)):
    df_4['Team'][i] = df['Club'][i]
    df_4['Expenditure'][i] = total_expenditure(df, df['Club'][i])
    df_4['Performance'][i] = df_3.loc[df_3['Team'] == df['Club'][i], 'Performance'].values[0]
    
# rank the teams by performance with N/A being the worst
df_4['Performance'] = df_4['Performance'].replace('N/A', 0)
df_4['Performance'] = df_4['Performance'].astype(int)
df_4 = df_4.sort_values(by=['Performance'], ascending=False)
df_4 = df_4.reset_index(drop=True)

df_4



Team Expenditure  Performance
0           Manchester City      1030.0           24
1                 Liverpool       674.6           18
2                   Chelsea      1490.0           14
3         Manchester United      984.98           12
4         Tottenham Hotspur       655.8           12
5                   Arsenal      839.56            4
6            Leicester City      477.75            3
7           West Ham United       600.7            3
8                   Burnley      181.29            2
9            Crystal Palace      223.02            1
10   Brighton & Hove Albion      374.59            1
11              Southampton      409.05            1
12         Newcastle United       534.0            1
13                  Everton      615.07            1
14              Aston Villa      512.69            0
15             Leeds United      372.81            0
16                   Fulham      303.42            0
17              Bournemouth      276.55            0
18         NottinghamForest      249.07            0
19  Wolverhampton Wanderers      556.36            0
20                  Watford      221.81            0
21         Sheffield United      150.34            0
22                Brentford      148.02            0
23     West Bromwich Albion      142.31            0
24                StokeCity      134.15            0

Based on the dataframe above it is clear to see that while spending a lot on transfers can help a clubs chances of succeeding, it does not guarentee it. 

The top 5 teams in terms of spending are Manchester City, Manchester United, Chelsea, Arsenal and Liverpool. 4 of those teams were also in the top 5 in terms of performance. City scored both the highest in spending and in performance. The lowest spender that received any points was Burnley, spending 181.29 million.

The top of this dataframe shows a pretty good indicator that if a team spends a lot, they will perform well, but things get much more unclear when we look at the bottom of the table. Many teams spent close to 1/2 a billion dollars over the last 5 years, but only received 1 point or 0.

It is intersting to note that Everton, the team with the 6th highest spending only received 1 point for never being relegated. Fulham spent 303 million dollars over the five years and received no points, but teams such as Burnley and Crystal Palace spent less while still remaining in the EPL. Some other interesting notes is that Wolverhampton does not have any points because they got promoted in 2018, but they have spent 556.36 million dollars over the last 5 years and have not been relegated since being promoted.

The data shows that spending a lot on transfers does not guarentee success, but it does help. The best way to spend money on transfers is to spend it on players that will help the team perform well. If a club wants to win the league, they'll have to fight with teams that have spend close to 1 billion dollars every 5 years. In the middle of the table, it is clear that how much you spend does not affect the performance of the team. Even spending a lot of money does not guarentee that the club will be in the EPL for a significant amount of time.
